### PR TITLE
Elaborate text-to-speech requirements on Linux

### PR DESCRIPTION
**Pull request by nightblade9**

### CODE REVIEW

**Comment by nightblade9** [comment]
Not sure how this is style-wise. I tried to avoid passive voice ("will not work if this is not installed") and erred on the side of redundancy/repetition to avoid any potential misunderstandings about how this works.

File: doc/classes/DisplayServer.xml [hunk 1]
```diff
@@ -984,6 +984,8 @@
 				- [code]name[/code] is voice name.
 				- [code]id[/code] is voice identifier.
 				- [code]language[/code] is language code in [code]lang_Variant[/code] format. [code]lang[/code] part is a 2 or 3-letter code based on the ISO-639 standard, in lowercase. And [code]Variant[/code] part is an engine dependent string describing country, region or/and dialect.
+				Note that, on Linux, Godot depends on system-dependent libraries for text-to-speech functionality. If they are not present, this method will return an empty list. This applies to both Godot users on Linux, as well as end-users on Linux running Godot games that use text-to-speech.
```
Comment (realkotob) [inline doc/classes/DisplayServer.xml:987]: I would suggest changing "system-dependent libraries" to "system libraries" for clarity.

**Comment by YuriSizov** [comment]
Don't we rely on system libraries on every system, not just on Linux?

Also, this should be one paragraph, not two. Each new line is a new paragraph, and here you're still describing the same thing.

**Comment by akien-mga** [comment]
> Don't we rely on system libraries on every system, not just on Linux?

That's correct. The main difference is that on a modern Windows or macOS system the TTS libraries are installed by default, so it Just Works™. While on Linux it's not guaranteed, it's often something you need to install manually when you need it (which IMO is a very bad approach to accessibility but hey, Linux isn't always the best :P).

**Comment by YuriSizov** [comment]
Makes sense. So then I would reword this note a bit, so it states that we depend on optional system libraries which may not be installed from the beginning.

**Comment by nightblade9** [comment]
Updated, thanks for clarifying. I personally feel like the example should be a second paragraph for clarity, but this works for me too.

Honestly, I really think this information doesn't belong (just) here, but in a "tutorial" style page about TTS. There are enough caveats and things to understand, and it sounds like most of us just want TTS to work, and don't really know (or care) about how the underlying mechanisms fit together.

Plus, I think there are some "best practices" about using TTS (e.g. provide the user options to select voices and customize the speech rate) that don't fit in a technical class document, but would be invaluable to anyone implementing this, in terms of providing a good user experience.

**Comment by Calinou** [comment]
I think we can have a generic note about TTS requiring libraries, but the class reference shouldn't mention specific Linux distributions or the packages to install. This should be done in a dedicated documentation page instead, which you can link to players.

**Comment by nightblade9** [comment]
Okay, I have removed the note/example about Arch. I'll circle around and add a link to the tutorial once I write it. Is this now good to go @Calinou ?

I also noticed one of the CI actions failed (the one that checks documentation), but I was unable to determine what the fix is based on the build output. Can you please provide some direction?

**Comment by nightblade9** [comment]
Tutorial's here: https://github.com/godotengine/godot-docs/pull/6783

If that gets merged first, I'll update this PR with a link.

**Comment by nightblade9** [comment]
CI is passing now. What's the next step to get this merged?

**Comment by YuriSizov** [comment]
> CI is passing now. What's the next step to get this merged?

You don't need to do anything for now, it needs to get reviewed and approved by a maintainer.

**Review by akien-mga — APPROVED** [review]
I'd change "distros" to "distributions", otherwise this looks good to me.

**Comment by nightblade9** [comment]
> I'd change "distros" to "distributions", otherwise this looks good to me.

Done, thanks for the feedback.

**Comment by akien-mga** [comment]
Thanks!

**Comment by nightblade9** [comment]
Thank you @akien-mga ! It's a very exciting historical moment for me :)
